include/stumpless/error.h

```diff
@@ -30,85 +30,100 @@
 extern "C" {
 #  endif
 
+
 /**
- * An identifier of the types of errors that might be encountered.
- *
- * Note that the same error may be encountered in different contexts with a
- * different ultimate cause. To completely diagnose a problem, you will need
- * to look at more than just the error id.
+ * A macro to implement the enum values && enum names string array.
  */
-enum stumpless_error_id {
-/**
+#define FOREACH_ERROR(ERROR)	\
+/** \var STUMPLESS_ADDRESS_FAILURE
  * A provided network address was not valid.
```

Comment:
For all multi-line comments within the macro expansion to render properly in doxygen, you will need to remove the asterisks at the beginning of the lines  (for example, lines 39, 40, and 41 here, lines 49-51, etc.)